### DISCUSSION TITLE
The command "PATH=$PATH:/usr/local/bin NODE_PATH=`npm root -g` node

### DISCUSSION THREAD

**Discussion by ousid** [spatie/laravel-sitemap#315] [Q&A]
when i tried to generate a sitemap in localhost it gives me an empty sitemap.

```php
    $path = public_path('/sitemap.xml');
    SitemapGenerator::create(env('APP_URL'))->writeToFile($path);
```
when upload the file into web server and tried to generate the sitemap it gives me this error:

```
The command "PATH=$PATH:/usr/local/bin NODE_PATH=`npm root -g` node '/path/to/project/vendor/spatie/browsershot/src/../bin/browser.js' '{"url":"https:\/\/coderflex.com\/","action":"content","options":{"args":[],"viewport":{"width":800,"height":600}}}'" failed. Exit Code: 1(General error) Working directory: /path/to/project/public Output: ================ Error Output: ================ internal/modules/cjs/loader.js:960 throw err; ^ Error: Cannot find module 'puppeteer' Require stack: - /path/to/project/vendor/spatie/browsershot/bin/browser.js at Function.Module._resolveFilename (internal/modules/cjs/loader.js:957:15) at Function.Module._load (internal/modules/cjs/loader.js:840:27) at Module.require (internal/modules/cjs/loader.js:1019:19) at require (internal/modules/cjs/helpers.js:77:18) at Object.<anonymous> (/path/to/project/vendor/spatie/browsershot/bin/browser.js:1:19) at Module._compile (internal/modules/cjs/loader.js:1133:30) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10) at Module.load (internal/modules/cjs/loader.js:977:32) at Function.Module._load (internal/modules/cjs/loader.js:877:14) at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12) { code: 'MODULE_NOT_FOUND', requireStack: [ '/path/to/project/vendor/spatie/browsershot/bin/browser.js' ] }
```

any ideas?

**Comment by VolodyaP** [comment]
Any updates?

**Comment by ousid** [comment]
@VolodyaP  No Updates so far.

**Comment by amir94rp** [comment]
do you solve it?

**Comment by ousid** [comment]
not yet :)

**Comment by VolodyaP** [comment]
Fixed.
My sitemap.php config file: 

'guzzle_options' => [
        // adding this option will ignore errors generated by self
        // signed https certs

        RequestOptions::COOKIES => true,
        RequestOptions::CONNECT_TIMEOUT => 3,
        RequestOptions::TIMEOUT => 5,
        RequestOptions::ALLOW_REDIRECTS => false,
        RequestOptions::VERIFY => false,
        RequestOptions::DELAY => 0,
        RequestOptions::SYNCHRONOUS => false,
    ],

    'execute_javascript' => true,

**Comment by amir94rp** [comment]
i used your config and still the same error

**Comment by rgbjay** [comment]
Hey guys not sure if this helps, was having the same issue, but I got it working for me using the below (manually setting the path for Node, NPM and Chromium). Using Laravel 8 and Homestead.

```
$browsershot = (new \Spatie\Browsershot\Browsershot)
    ->setNodeBinary('/usr/bin/node')
    ->setNpmBinary('/usr/bin/npm')
    ->setChromePath('/usr/bin/chromium-browser');
SitemapGenerator::create($site_url)
     ->configureCrawler(function (Crawler $crawler) use ($browsershot) {
         $crawler->executeJavaScript()->setBrowsershot($browsershot);
     })
    ->getSitemap()
    ->writeToDisk('local', 'sitemap.xml');
```

**Comment by crmarin** [comment]
I found the laravel user isn't the current user to run node scripts, I don't know if could be root or www-data. This for me will be my discover
I solve easy, moving the /usr/local/bin/node inside the laravel project, for example to /scripts, then updated my path environment to this new path, it works
